### ISSUE TITLE
Replace Raffle Rounds with upgraded Conferences functionality  

### ISSUE THREAD

**Issue by F3PiX** [rails-girls-summer-of-code/rgsoc-teams#751]
Conferences have a `round` attribute, that indicates if they are in the first, second (or more?) raffle rounds where participants can win a ticket.  

### Problem
As I understand it, this feature is not fully implemented.
- In the dropdown menu (in users/:id/edit) where users can choose their fav conference, all the [conferences in the current season](https://github.com/rails-girls-summer-of-code/rgsoc-teams/blob/master/app/views/users/edit.html.slim#L65) appear. I think last year, an orga would manually add the raffle round number to the name of the conf, and the students need to choose conferences that starts with that number. 

- AFAIK there is no way for orga's to mark when the first round is finished. If there were, we could make better use of existing functionality; not only the dropdown menu, but also the (sequence in) the table where conferences are listed, the number of conferences a user can choose per round (see #575 ) etc.

### Solution
-[ ] Add a way to mark the current raffle round, or mark just which raffle round is coming up next.

### Implementation
This is what I am not sure about. Should the end of a raffle round be part of the season? (Other than other season phases, the dates for the raffle are not known upfront.)  
The simplest way seems to be: just add a checkbox or radio button somewhere, that orga's can check to indicate what raffle round is currently relevant. 

What shall we do about this?

**Comment by F3PiX** [comment]
@anikalindtner Can you shed some light on the conference raffle operations and how we can improve that?  ^

**Comment by carpodaster** [comment]
@F3PiX can we make the conference raffle round an attribute of `Season`? Not that it has to switch automatically by date, but at least that way an organizer can update the global state of the app's behaviour without us having to trigger a `git commit` + deployment.

**Comment by mkalininait** [comment]
@F3PiX @carpodaster The thing is that we don't do raffles anymore. We pick up friendly conferences, ask for students' preferences within that list and then just try to get them tickets to the preferred conference(s). Last year we used this new process and it worked out well (imo), but we didn't use the Teams App at all. I would really love to see how we can change the Teams app this year to fit the new process better.
The process is documented here: https://github.com/rails-girls-summer-of-code/organization/wiki/Processes:-Conferences Do you have access to it?

**Comment by F3PiX** [comment]
Aha! That _is_ different from what I envisioned. I don't have access. Can you share it with me, one way or another?

**Comment by mkalininait** [comment]
@F3PiX I've sent you the document on Slack; you can also ping me here and there regarding any details!

**Comment by mkalininait** [comment]
@F3PiX Just found this issue: https://github.com/rails-girls-summer-of-code/rgsoc-teams/issues/517 Did you see it? 

Here is my thinking on how we could handle conferences in the Teams app (and it basically repeats the ideas expressed in that conference document I sent to you).
1) Use the Teams app to communicate with the students:
 - Get conferences of their preference;
 - Get confirmation of their attendance (once we have tickets from conference organisers).
2) Don't use the Teams app as a database of conferences; it's much easier and more convenient to use Google Spreadsheets for this.
3) Get rid of the raffle functionality.

Below is the conference functionality I could imagine in the Teams app, not limited by our dev capabilities yet :D Feedback is welcome! Or we can move it to a separate issue if it makes more sense.

**Conference block on the Team's edit page:**
----
**CONFERENCE PREFERENCES** (this part is visible to the students)
Please fill out your preferences regarding the conference you would like to attend as part of RGSoC.
 * (Dropdown) Conference (primary choice)
 * (Dropdown) Conference (secondary choice)
 * (Button) I want to suggest another conference! // opens the form "New conference"
 * [ ] We would like to give a lightning talk if possible
 * Comments (e.g. "We would attend conference X only in they provide scholarships / have a LT session" or "We have already applied for the conference X ourselves" etc.)
 * [ ] We understand that RGSoC doesn't guarantee tickets to the conferences of our choice.
 * [ ] We understand that all travel and accommodation costs will have to be covered by us; hereby we confirm that we are choosing conferences which we can afford.

**OFFERED CONFERENCES** (this part is visible to the orga)
 * (Dropdown) Conference 1
 * (Button) Add another conference // adds another dropdown

On the team's page students can press one of two links next to the offered conferences: 
 * Yes, I can attend 
 * No, I cannot attend


**Import/export:**
----
 * It would be great to implement a csv export of all conference data (e.g. conferences#current).
 * It would be also cool if we could import a csv list of conferences to the Teams app (I mean just conference data, without any associations with the teams -- to import the initial list of conferences we want to suggest to the students).

**Comment by F3PiX** [comment]
@mkalininait  Thanks for the overview. 
So, the conf database stays in Google docs.
 
For the part 'Communicating with the students':  How about we split the thing up in a kind of 'user stories', so we can differentiate between implementation details, design and functionality. Also, we can easily make the user stories actionable, and we can work in iterations. 
Are you okay with that? 

Let's collect the user stories in this issue, because this issue gives the context.  
As an example for the first user stories:

1) As a conference-orga, I want to import a 'long-list' of the conferences we want to suggest to the students. (Timeline: Around the start of the season.) 
 
2) As a student, I can choose my primary and secondary choices from the conferences longlist.

Wdyt?

**Comment by mkalininait** [comment]
@F3PiX Sounds like a good idea! I'll prepare the user stories and put them here.

**Comment by F3PiX** [comment]
Cool! Now we are going all scrumm-y ;-)
To speed up things, I think we don't need to have more than two user stories upfront. I'll start with adding an issue for story #1.

**Comment by mkalininait** [comment]
@F3PiX Here are the user stories:

1. As a RGSoC orga, I want to import a 'long-list' of the conferences we want to suggest to the teams. (Timeline: Around the start of the season.)

1. As a student, I can choose my team's primary and secondary choices from the conferences longlist.

1. As a student, I can add a cool conference which is missing in the longlist; I can choose it as my team's primary or secondary conference then.

1. As a student, I want to provide additional information to my team's conference choices: conditions / limitations / wishes etc., as well as our potential wish to give a lightning talk.

1. As an RGSoC orga, I want the students to be aware of the terms on which we hand out free conference tickets and confirm their obedience.

1. As an RGSoC orga, I want to export teams' conference preferences as a csv.

1. As an RGSoC orga, I want to see team's conference preferences and assign one or multiple conferences from the longlist which are actually willing to give them free tickets (with an optional comment).

1. As an RGSoC orga, I want to know if the students accept our offer.

**Comment by F3PiX** [comment]
Ooops. I thought we agreed to make the issues after the previous user stories are done.

**Comment by mkalininait** [comment]
@F3PiX Sorry, I understood it as you wanted to start asap, not as you actively don't want the issues to be in GH.
I talked to @carpodaster , he said that he would take a little task. I'm also thinking of taking a task myself. The thing is that we will start working on the conferences soon (~ mid-July), and it would be great to have the conference features then. And thus, I created the issues. If it doesn't work out (e.g. we don't have enough dev power), that's fine too, I can run the conference process without the Teams app this year again.
